src/components/ApplicationMenu/FramelessMenu.js

```diff
@@ -0,0 +1,292 @@
+// @flow
+import React, { Component } from 'react';
+import styled from 'styled-components';
+
+import Logo from '../Logo';
+
+import Icon from 'react-icons-kit';
+import { minus, minimize2, maximize2, x } from 'react-icons-kit/feather';
+import { COLORS, Z_INDICES } from '../../constants';
+import { remote } from 'electron';
+import {
+  execute,
+  getDefaultLabel,
+  getDefaultAccelerator,
+} from './menu-item-roles';
+import { isMac } from '../../services/platform.service';
+
+type MenuItem = {
+  role?: string,
+  type?: string,
+  click?: () => void,
+  label?: string,
+  accelerator?: string,
+};
+
+type Props = {
+  template: any,
+};
+type State = {
+  windowIsMaximized: boolean,
+  altIsPressed: boolean,
+};
+
+class FramelessMenu extends Component<Props, State> {
+  // is `context: any` the correct way to do this?
+  constructor(props: Props, context: any) {
+    super(props, context);
+
+    this.state = {
+      windowIsMaximized: remote.getCurrentWindow().isMaximized(),
+      altIsPressed: false,
+    };
+  }
```

Comment:
I don't think you need the constructor at all :) you should just be able to do:

```js
class FramelessMenu {
  state = {
    windowIsMaximized: remote.getCurrentWindow().isMaximized(),
    altIsPressed: false,
  }
}
```

(and to answer your question, I believe you'd only need to specify context if you needed it within the constructor)